barretenberg/cpp/src/barretenberg/dsl/acir_format/multi_scalar_mul.cpp

```diff
@@ -0,0 +1,48 @@
+#include "multi_scalar_mul.hpp"
+#include "barretenberg/dsl/types.hpp"
+#include "barretenberg/ecc/curves/bn254/fr.hpp"
+#include "barretenberg/ecc/curves/grumpkin/grumpkin.hpp"
+#include "barretenberg/plonk_honk_shared/arithmetization/gate_data.hpp"
+#include "barretenberg/stdlib/primitives/biggroup/biggroup.hpp"
+
+namespace acir_format {
+
+template <typename Builder> void create_multi_scalar_mul_constraint(Builder& builder, const MultiScalarMul& input)
+{
+    using cycle_group_ct = bb::stdlib::cycle_group<Builder>;
+    using cycle_scalar_ct = typename bb::stdlib::cycle_group<Builder>::cycle_scalar;
+    using field_ct = bb::stdlib::field_t<Builder>;
+
+    std::vector<cycle_group_ct> points;
+    std::vector<cycle_scalar_ct> scalars;
+
+    for (size_t i = 0; i < input.points.size(); i += 2) {
+        // Instantiate the input point/variable base as `cycle_group_ct`
+        auto point_x = field_ct::from_witness_index(&builder, input.points[i]);
+        auto point_y = field_ct::from_witness_index(&builder, input.points[i + 1]);
+        cycle_group_ct input_point(point_x, point_y, false);
+
+        // Reconstruct the scalar from the low and high limbs
+        field_ct scalar_low_as_field = field_ct::from_witness_index(&builder, input.scalars[i]);
+        field_ct scalar_high_as_field = field_ct::from_witness_index(&builder, input.scalars[i + 1]);
+        cycle_scalar_ct scalar(scalar_low_as_field, scalar_high_as_field);
+
+        // Add the point and scalar to the vectors
+        points.push_back(input_point);
+        scalars.push_back(scalar);
+    }
+
+    // Call batch_mul to multiply the points and scalars and sum the results
+    auto output_point = cycle_group_ct::batch_mul(scalars, points);
```

Comment:
@TomAFrench this is not calling [the function you sent me](https://github.com/AztecProtocol/aztec-packages/blob/86439222226e00f64945e54b2bf1c89c3d87e7ee/barretenberg/cpp/src/barretenberg/stdlib/primitives/biggroup/biggroup.hpp#L190) but instead it's calling [this one](https://github.com/AztecProtocol/aztec-packages/blob/a73e2ff9bf35f7e837faa01076ded83b0c762803/barretenberg/cpp/src/barretenberg/stdlib/primitives/group/cycle_group.cpp#L1187).

The one you provided me with is not available on cycle_group class so I think I did the correct thing here. But I suck at C++ so please confirm.

Reply:
Yep, that looks like the correct one, sorry for the runaround